### ISSUE TITLE
NBT tags don't work

### ISSUE THREAD

**Issue by LgndTheDredgen** [300f20t/Recipe-Generator#7]
The recipes generate fine, love it for fast and easy recipe generating since I suck at it, but unfortunately it doesn't include the nbt tag so I can't separate nor specify a specific item to use
Here's some in game examples of me trying it in minecraft chat form and it's still not giving me the correct items
![Screenshot 2024-11-15 012527](https://github.com/user-attachments/assets/494b482d-0d01-4b37-a66e-39cad12f1c26)
![image_2024-11-15_012659140](https://github.com/user-attachments/assets/60c0881d-983b-41ae-81c4-278fe59ace7a)

**Comment by 300f20t** [comment]
I will try to fix it.

**Comment by LgndTheDredgen** [comment]
> I will try to fix it.

there's also an issue when making the custom recipes,
it won't use a specific pearl type if you include it as an ingredient on a crafting table, it always comes out as an Error Pearl like shown in the image

**Comment by 300f20t** [comment]
this is related to NBT tags, so far the problem has turned out to be more difficult than I expected, since there is no built-in ability to get all NBT tags of an item

**Comment by LgndTheDredgen** [comment]
> this is related to NBT tags, so far the problem has turned out to be more difficult than I expected, since there is no built-in ability to get all NBT tags of an item

does this mean currently, if I were to use a pearl like the slime pearl as an ingredient to craft it into a blaze pearl, it just isn't possible

**Comment by 300f20t** [comment]
no, it should work, but my mod doesn't support NBT tags yet. I'm still trying to figure out a way to add this.

**Comment by LgndTheDredgen** [comment]
> no, it should work, but my mod doesn't support NBT tags yet. I'm still trying to figure out a way to add this.

my bad for asking a lot from you, I'm just confused how to do so cause whenever I try to input it, it comes out as a generic Gate Pearl item, rather than a specific one like the Slime Pearl I'm trying to use as an ingredient
